chia/consensus/block_creation.py

```diff
@@ -213,8 +213,15 @@ def create_foliage(
         additions_root = addition_merkle_set.get_root()
         removals_root = removal_merkle_set.get_root()
 
-        generator_hash = solution_program.get_tree_hash() if solution_program is not None else bytes32([0] * 32)
-        generator_refs_hash = bytes32([1] * 32)
+        generator_hash = bytes32([0] * 32)
+        if solution_program is not None:
+            generator_hash = std_hash(solution_program)
+
+        generator_refs_hash = bytes32([0] * 32)
+        if generator_block_heights_list not in (None, []):
+            generator_ref_list_bytes = b"".join([(i).to_bytes(4, byteorder='big') for i in generator_block_heights_list])
```

Comment:
Please take a look at this method of List[uint32] -> bytes conversion

Reply:
looks good, anything specific that was worrying you here?

Reply:
It seems weird to have this serialization logic here. Any way you can use `uint32`?

Reply:
Maybe try `b''.join(bytes(uint32(value)) for value in list_of_values)`